mission/FSM/docking_cpp/src/docking.cpp

```diff
@@ -0,0 +1,574 @@
+                    feedback->feedback.surge, feedback->feedback.pitch,
+                    feedback->feedback.yaw);
+    }
+}
+
+GoAboveDockingStationState::GoAboveDockingStationState(
+    std::shared_ptr<yasmin::blackboard::Blackboard> blackboard)
+    : yasmin_ros::ActionState<ReferenceFilterWaypoint>(
+          blackboard->get<std::string>("reference_filter_action"),
+          std::bind(&GoAboveDockingStationState::create_goal_handler, this, _1),
+          std::bind(&GoAboveDockingStationState::response_handler,
+                    this,
+                    _1,
+                    _2),
+          std::bind(&GoAboveDockingStationState::print_feedback,
+                    this,
+                    _1,
+                    _2)) {};
+
+ReferenceFilterWaypoint::Goal GoAboveDockingStationState::create_goal_handler(
+    std::shared_ptr<yasmin::blackboard::Blackboard> blackboard) {
+    auto goal = ReferenceFilterWaypoint::Goal();
+
+    auto docking_offset_goal =
+        blackboard->get<PoseStamped>("docking_offset_goal");
+    goal.goal = docking_offset_goal;
+
+    RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "Goal sent to action server:");
+    RCLCPP_INFO(rclcpp::get_logger("rclcpp"),
+                "  Position: x = %f, y = %f, z = %f",
+                docking_offset_goal.pose.position.x,
+                docking_offset_goal.pose.position.y,
+                docking_offset_goal.pose.position.z);
+    RCLCPP_INFO(rclcpp::get_logger("rclcpp"),
+                "  Orientation: x = %f, y = %f, z = %f, w = %f\n",
+                docking_offset_goal.pose.orientation.x,
+                docking_offset_goal.pose.orientation.y,
+                docking_offset_goal.pose.orientation.z,
+                docking_offset_goal.pose.orientation.w);
+
+    return goal;
+}
+
+std::string GoAboveDockingStationState::response_handler(
+    std::shared_ptr<yasmin::blackboard::Blackboard> blackboard,
+    ReferenceFilterWaypoint::Result::SharedPtr response) {
+    RCLCPP_INFO(rclcpp::get_logger("rclcpp"),
+                "Response received from action server:");
+    RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "  Success: %s\n",
+                response->success ? "true" : "false");
+
+    blackboard->set<bool>("is_docked", response->success);
+
+    if (blackboard->get<bool>("is_docked")) {
+        return yasmin_ros::basic_outcomes::SUCCEED;
+    } else {
+        return yasmin_ros::basic_outcomes::CANCEL;
+    }
+}
+
+void GoAboveDockingStationState::print_feedback(
+    std::shared_ptr<yasmin::blackboard::Blackboard> blackboard,
+    std::shared_ptr<const ReferenceFilterWaypoint::Feedback> feedback) {
+    Pose current_pose = Pose();
+    current_pose.position.x = feedback->feedback.x;
+    current_pose.position.y = feedback->feedback.y;
+    current_pose.position.z = feedback->feedback.z;
+
+    blackboard->set<Pose>("current_pose", current_pose);
+
+    if (blackboard->get<bool>("print_debug")) {
+        RCLCPP_INFO(rclcpp::get_logger("rclcpp"),
```

Comment:
Same case as my comment above wrt log levels. Standard practice is SPDLOG_DEBUG / RCLCPP_DEBUG / etc... for debug statements, and setting the log level in main